test/fail_compilation/fail20547.d

```diff
@@ -0,0 +1,13 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/fail20547.d(11): Error: cannot create a `int[string]` with `new`
+fail_compilation/fail20547.d(12): Error: cannot create a `int[string]` with `new`
+---
+*/
+
+void main()
+{
```

Comment:
Please add the example from bugzilla.

```suggestion
{
    int[int] a = new int[int];
```

Reply:
Since this error is detected by the parser, dmd actually stops here and doesn't report errors for b and c when a is present.  I added another test for that example.

Reply:
Thats odd, the code below shows an error concerning the last line while hiding the two previous ones ...

```
void main()
{
    int[string] b = new int[string];
    int[string] c = new typeof(b);
    int[int] a = new int[int];
}
```

Reply:
You're right, the first lines are resolved in semantic (probably because `string` is an alias instead of a keyword).

Reply:
The additional file for `int[int]` is actually not needed. I missed the change in `misc_parser_err_cov1.d` which already tests an equivalent case.
Sorry for the noise.

Reply:
No worries, taken out.